tests/telemetry_test.py

```diff
@@ -3664,3 +3669,78 @@ def test_stores_nothing_if_no_data_path(self, run_subprocess, metrics_store_clus
         self.assertEqual(0, run_subprocess.call_count)
         self.assertEqual(0, metrics_store_cluster_value.call_count)
         self.assertEqual(0, get_size.call_count)
+
+
+class TestMasterNodeStats:
+    @mock.patch("esrally.telemetry.SamplerThread.start")
+    @mock.patch("esrally.telemetry.SamplerThread.finish")
+    def test_record_start_and_end(self, start, finish, monkeypatch):
+        record_count = 0
+
+        def record(self):
+            nonlocal record_count
+            record_count += 1
+
+        monkeypatch.setattr(telemetry.MasterNodeStatsRecorder, "record", record)
+
+        clients = {"default": Client(), "cluster_b": Client()}
+        cfg = create_config()
+        metrics_store = metrics.EsMetricsStore(cfg)
+        telemetry_params = {"master-node-stats-sample-interval": 0.001}
+        t = telemetry.MasterNodeStats(telemetry_params, clients, metrics_store)
+        t.on_benchmark_start()
+        t.on_benchmark_stop()
+
+        # the two clients record start and stop
+        assert record_count == 4
```

Comment:
For what it's worth I tried using `@mock.patch` and `call_count` here but I was only seeing 2 calls, not 4. I still don't know why. I ended up switching to the pytest style which is more verbose but less magical.